ui/src/app/pages/login/account-creation/account-creation.spec.tsx

```diff
@@ -103,128 +97,116 @@ test.each([
   let testInput = fp.repeat(50, 'abc');
   testInput = testInput + ' abc';
-  wrapper
-    .find('input#username')
-    .simulate('change', { target: { value: testInput } });
-  expect(wrapper.exists('#usernameError')).toBeTruthy();
+  await user.type(container.querySelector('input#username'), testInput);
+  expect(container.querySelector('#usernameError')).not.toBeNull();
 });
 
-it('should handle username validity length less than 3 characters', () => {
-  const wrapper = component();
-  expect(wrapper.exists('#username')).toBeTruthy();
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
-  wrapper.find('input#username').simulate('change', { target: { value: 'a' } });
-  expect(wrapper.exists('#usernameError')).toBeTruthy();
+it('should handle username validity length less than 3 characters', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('#username')).not.toBeNull();
+  expect(container.querySelector('#usernameError')).toBeNull();
+  await user.type(container.querySelector('input#username'), 'a');
+  expect(container.querySelector('#usernameError')).not.toBeNull();
 });
 
-it('should handle username validity if name is valid', () => {
-  const wrapper = component();
-  expect(wrapper.exists('#username')).toBeTruthy();
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
-  wrapper
-    .find('input#username')
-    .simulate('change', { target: { value: 'username' } });
-  expect(wrapper.exists('#usernameError')).toBeFalsy();
+it('should handle username validity if name is valid', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
+  expect(container.querySelector('#username')).not.toBeNull();
+  expect(container.querySelector('#usernameError')).toBeNull();
+  await user.type(container.querySelector('input#username'), 'username');
+  expect(container.querySelector('#usernameError')).toBeNull();
 });
 
-it('should display characters over message if research purpose character length is more than 2000', () => {
-  const wrapper = component();
+it('should display characters over message if research purpose character length is more than 2000', async () => {
+  const user = userEvent.setup();
+  const { container } = component();
 
-  const areaOfResearchTextBox = wrapper.find('[id="areaOfResearch"]');
-  expect(getCharactersLimitProps(wrapper).children).toEqual(
-    '2000 characters remaining'
+  const areaOfResearchTextBox = container.querySelector(
+    'textarea#areaOfResearch'
   );
+  expect(screen.queryByText('2000 characters remaining')).not.toBeNull();
 
   let testInput = fp.repeat(2000, 'a');
-  areaOfResearchTextBox
-    .find('textarea#areaOfResearch')
-    .simulate('change', { target: { value: testInput } });
-  expect(getCharactersLimitProps(wrapper).children).toEqual(
-    '0 characters remaining'
-  );
-  expect(getCharactersLimitProps(wrapper).style.color).toBe(colors.danger);
```

Comment:
A couple test checked the color of the text. I decided to remove them after spending ~15 minutes trying (and failing) to figure out how to check it. I was able to find the RGB value, but not the hex value (what `colors.danger` is). I suspect there's a more accessible way to mark text as "danger."